### PR TITLE
Implement Cosmos DB for the Devices Table

### PR DESCRIPTION
**Pull request by alexfca**
This PR uses Cosmos DB for the device_devices SQL table.
The changes were similar to the last set, so no major changes.
The Config for the Tenant (Database+Container) details are now included in the ConnectionString in the YAML.
As such, the conn string for Cosmos requires;
- "cosmosdb:" - This is the prefix that determines the system should use Cosmos (eg sqllite uses "file:")
- "AccountEndpoint" - This is the Cosmos URL - eg "https://localhost:8081/"
- "AccountKey" - This is the Cosmos Key - eg "C2y6yDjf5/R+ob0N.... =="
- "DatabaseName" - This is the Cosmos Database to store the data - eg "safezone_local"
- "ContainerName" - This is the Cosmos Container to store the data - eg "criticalarc.com"
An example conn string is
`"cosmosdb:AccountEndpoint=https://localhost:8081/;AccountKey=C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==;DatabaseName=safezone_local;ContainerName=criticalarc.com;"
`

The Devices are used in the UI where the user can manage Sessions and Devices;
Devices
image

and Sessions
image

The data is stored in a similar way to the other objects;
image

- Add cosmos devices
- Use the ConnectionString in the YAML to include the Tenant
- Revert all other non implemented tables back to use SQLLite3

### CODE REVIEW

File: internal/cosmosdbutil/connection.go [hunk 1]
```diff
@@ -1,22 +1,50 @@
...
+			itemSplit := strings.SplitN(item, "=", 2)
+			connectionItems[itemSplit[0]] = itemSplit[1]
+		}
 	}
 	return connectionItems
-}
\ No newline at end of file
+}
+
+func GetCosmosConnection(d *config.DataSource) cosmosdbapi.CosmosConnection {
```
Comment (jahmai-ca) [inline internal/cosmosdbutil/connection.go:33]: lower case get?

File: internal/cosmosdbutil/connection.go [hunk 2]
```diff
@@ -1,22 +1,50 @@
...
+	connString := getConnectionString(d)
+	connMap := getConnectionProperties(string(connString))
+	accountEndpoint := connMap[accountEndpointName]
+	accountKey := connMap[accountKeyName]
+	return cosmosdbapi.GetCosmosConnection(accountEndpoint, accountKey)
+}
+
+func GetCosmosConfig(d *config.DataSource) cosmosdbapi.CosmosConfig {
```
Comment (jahmai-ca) [inline internal/cosmosdbutil/connection.go:41]: lower case get?

File: internal/cosmosdbutil/connection.go (hunk 1 above)
Comment (jahmai-ca) [reply internal/cosmosdbutil/connection.go:33]: Most of the functions seem to be PascalCase, but `getConnectionString` is camelCase. Is that normal in Go?

Comment (alexfca) [reply internal/cosmosdbutil/connection.go:33]: Yeah, its a little weird, but camelCase is internal and PascalCase is public in GO

> In Go this is not a matter of style, it is a matter of syntax.
> 
> Exported names (that is, identifiers that can be used from a package other than the one where they are defined) begin with a capital letter. Thus if your method is part of your public API, it should be written:
> 
> WriteToDB
> 
> but if it is an internal helper method it should be written:
> 
> writeToDB

https://stackoverflow.com/questions/38616687/which-way-to-name-a-function-in-go-camelcase-or-semi-camelcase
https://golang.org/ref/spec#Exported_identifiers

Comment (jahmai-ca) [reply internal/cosmosdbutil/connection.go:33]: Got it thanks

**Review by jahmai-ca — APPROVED** [review]

**Review by jaredjn — APPROVED** [review]

**Pull request merged**